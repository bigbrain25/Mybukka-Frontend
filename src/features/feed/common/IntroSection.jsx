import React from 'react';

import PropTypes from 'prop-types';

import Container from 'Components/container';
import Navbar from 'Components/navbar';

import './introsection.scss';

const bgImage =
  'https://res.cloudinary.com/dn93xk5ni/image/upload/v1550614423/basil-cheese-close-up-1438672_qk0t5t.jpg';

const LargeTextSection = () => (
  <Container classNames="intro-text pl-2">
    <h2 className="larger-text text-left">You Crave.</h2>
    <h2 className="larger-text text-left">We get it.</h2>
  </Container>
);

const IntroSection = ({ push }) => (
  <div
    className="feed-intro-section img-fluid intro-section-fixed-height"
    style={{ backgroundImage: `url(${bgImage})` }}
  >
    <div className="intro-navbar-area">
      <Navbar push={push} bukka />
    </div>
    <div className="fixed-top text-left" style={{ zIndex: '100' }}>
      <LargeTextSection />
    </div>
  </div>
);

export default IntroSection;

IntroSection.propTypes = {
  push: PropTypes.func.isRequired,
};
