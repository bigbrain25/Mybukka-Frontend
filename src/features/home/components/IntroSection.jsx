import React from 'react';
import PrimaryNavbar from 'Components/navbar/PrimaryNavbar';
import SearchLocation from '../common/SearchLocation';

import './introsection.scss';

const LargeTextSection = () => (
  <div className="container large-text-section">
    <h2 className="text-center larger-text">
      Anything, anytime, anywhere. We get it.
    </h2>
  </div>
);

const SmallTextSection = () => (
  <div className="container small-text-section">
    <h5 className="text-center small-text">
      Food, drinks and groceries available for delivery or pickup.
    </h5>
  </div>
);

const SearchLocationSection = () => (
  <div className="container search-location-section">
    <SearchLocation />
  </div>
);

const IntroSection = () => (
  <div className="intro-section">
    <PrimaryNavbar />
    <LargeTextSection />
    <SmallTextSection />
    <SearchLocationSection />
  </div>
);

export default IntroSection;
