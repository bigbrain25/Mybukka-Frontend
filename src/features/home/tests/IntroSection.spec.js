import React from 'react';

import IntroSection
  from '../components/IntroSection';

describe('Area to explore component', () => {
  const wrapper = shallow(<IntroSection push={jest.fn()} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
