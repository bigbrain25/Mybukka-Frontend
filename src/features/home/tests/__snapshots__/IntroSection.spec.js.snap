// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Area to explore component renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IntroSection />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <PrimaryNavbar />,
        <LargeTextSection />,
        <SmallTextSection />,
        <SearchLocationSection />,
      ],
      "className": "intro-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <PrimaryNavbar />,
          <LargeTextSection />,
          <SmallTextSection />,
          <SearchLocationSection />,
        ],
        "className": "intro-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
